### ISSUE TITLE
Failed to build `alsa-sys v0.1.2` - "failed to run pkg-config"

### ISSUE THREAD

**Issue by futurefarmer** [nannou-org/nannou#282]
Hi, Thanks for your work! Just trying to set up your framework but getting an error when trying to run the example.

Got this:

failed to run custom build command for `alsa-sys v0.1.2`
process didn't exit successfully: `/home/lynx/nannou/target/release/build/alsa-sys-955037715e9ddf22/build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: "Failed to run `\"pkg-config\" \"--libs\" \"--cflags\" \"alsa\"`: No such file or directory (os error 2)"', src/libcore/result.rs:997:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

**Comment by mitchmindtree** [comment]
Hey @futurefarmer, thanks for the issue!

To get ALSA working (the low-level audio API) you might need to ensure the alsa dev packages are installed on your system. Have you followed the linux steps [here](https://github.com/nannou-org/nannou#platform-specific-setup)? You might also need to install pkg-config too by the looks of it - I'll make sure we add that to the README too! Which distro of linux are you using btw?

**Comment by linusbeff** [comment]
You could also try `sudo apt install libasound2-dev` that fixed it for me.

**Comment by gaocegege** [comment]
Same issue here, solved by installing the pkg-config

```
apt install pkg-config
```

**Comment by mitchmindtree** [comment]
Closing thanks to nannou-org/guide#37

**Comment by yaikohi** [comment]
I know this thread is closed but I had followed all instructions in the docs but still had the issue.
 
For me it was solved by installing librust-alsa-sys-dev (`sudo apt install librust-alsa-sys-dev`).

[Github thread where I got the solution from](https://github.com/Spotifyd/spotifyd/issues/659#issuecomment-737730954).

Posting it here for others that potentially run into this as well. 

I am not very experienced but is this something that should be added to the nannou docs/linux setup section?
